web/server/codechecker_server/auth/cc_ldap.py

```diff
@@ -410,8 +410,10 @@ def get_groups(ldap_config, username, credentials):
                               user_dn_postfix_preference)
 
         group_pattern = ldap_config.get('groupPattern', '')
-        if user_dn and group_pattern == '':
-            # User found and there is no group membership pattern to check.
+        if user_dn and not group_pattern:
+            LOG.debug("User '%s' found but there is no group_pattern"
+                      " to check LDAP for group membership.",
+                      user_dn)
             return []
```

Comment:
I'm not sure if returning `[]` is the correct strategy here. `search_s()` on line 451 will still work even when group_pattern is `None`. According to the ldap docs:
> Changed in version 3.0: filterstr=None is equivalent to filterstr='(objectClass=*)'.

In this case I could change the patch to proceed and actually perform the search.

Reply:
@jimis I think we can keep the current behavior, less search requests will go to the ldap @csordasmarton, @bruntib  what do you think?

Reply:
@gyorb I think the same, that it is better to perform less search queries to the ldap if there is no need.